src/librustc_typeck/astconv.rs

```diff
@@ -1816,6 +1816,16 @@ impl<'o, 'gcx: 'tcx, 'tcx> dyn AstConv<'gcx, 'tcx> + 'o {
             hir::TyKind::Err => {
                 tcx.types.err
             }
+            hir::TyKind::CVarArgs => {
+                let va_list_did = match tcx.lang_items().va_list() {
+                    Some(did) => did,
+                    None => bug!("`va_list` lang item required for variadics"),
```

Comment:
Either use `span_bug!` with a useful span or just use `tcx.lang_items().va_list().expect("`va_list` lang item required for variadics")`

Also add a `#![no_core]` test without the `va_list` lang item but with a variadic function.

Reply:
you can take the span from the `ast_ty` argument of this function for the `span_bug`